src/PgsqlServer.cpp

```diff
@@ -0,0 +1,433 @@
+    _socket.write(parameter, strlen(parameter));
+}
+
+void PgsqlWorker::sendServerConfiguration(){
+        _socket.write((char*) RESPONSE_MESSAGE_KEY, sizeof(RESPONSE_MESSAGE_KEY));
+
+        //Space in start of configuration message
+        writeSizeOfBlock(sizeof(uint32_t) + _delim_len * 4);
+
+        _socket.write((char*) _delimeter, _delim_len);
+
+        _socket.write((char*) _delimeter, _delim_len);
+
+        _socket.write((char*) _delimeter, _delim_len);
+
+        _socket.write((char*) _delimeter, _delim_len);
+        //End of space in start message
+
+        sendParameter("application_name", "", "S");
+
+        _socket.write((char*) _delimeter, _delim_len);
+
+        sendParameter("client_encoding", "UTF8", "S");
+
+        _socket.write((char*) _delimeter, _delim_len);
+
+        sendParameter("DateStyle", "ISO, MDY", "S");
+
+        _socket.write((char*) _delimeter, _delim_len);
+
+        sendParameter("integer_datetimes", "on", "S");
+
+        _socket.write((char*) _delimeter, _delim_len);
+
+        sendParameter("IntervalStyle", "postgres", "S");
+
+        _socket.write((char*) _delimeter, _delim_len);
+
+        sendParameter("is_superuser", "on", "S");
+
+        _socket.write((char*) _delimeter, _delim_len);
+
+        sendParameter("server_encoding", "UTF8", "S");
+        
+        _socket.write((char*) _delimeter, _delim_len);
+
+        sendParameter("server_version", "9.6.5", "S");
+
+        _socket.write((char*) _delimeter, _delim_len);
+
+        sendParameter("session_authorization", "postgres", "S");
+
+        _socket.write((char*) _delimeter, _delim_len);
+
+        sendParameter("standard_conforming_strings", "on", "S");
+
+        _socket.write((char*) _delimeter, _delim_len);
+
+        sendParameter("TimeZone", "W-SU", "S");
+
+        _socket.write((char*) _delimeter, _delim_len);
+
+        _socket.write((char*) "K");
+        
+        //Not meaning end of configuration message
+        unsigned char end[] = { 0x00, 0x00, 0x00, 0x0C, 0x00, 0x00, 0x49, 0x4A, 0x01, 0x08, 0x39, 0xFF, 0x5A, 0x00, 0x00, 0x00, 0x05, 0x49};
+    
+        _socket.write((char*) end, sizeof(end));
+}
+
+void PgsqlWorker::writeCodeAnswer(const char* command, const char* status, const char * response_type){
+    uint32_t message_size = htonl(sizeof(uint32_t) + strlen(command) + strlen(status));
+    
+    _socket.write(COMMAND_KEY, sizeof(COMMAND_KEY)); //Code key
+    _socket.write((char*) &message_size, sizeof(message_size));
+    _socket.write(command, strlen(command));
+    _socket.write((char*) _delimeter, _delim_len);
+    _socket.write(status, strlen(status));
+
+    message_size = htonl(sizeof(uint32_t) + strlen(response_type));
+
+    _socket.write((char*) &message_size, sizeof(message_size));
+    _socket.write(response_type, strlen(response_type));
+}
+
+char* PgsqlWorker::readMessage(){
+    size_t size;
+    char* received_message;
+
+    size = getMessageSize();
+    received_message = new char[size];
```

Comment:
received_message will leak if _socket.read() throws an exception. It's better to provide a buffer on calling side, e.g. see Socket class interface.